src/main/java/com/gregtechceu/gtceu/common/machine/electric/WorldAcceleratorMachine.java

```diff
@@ -0,0 +1,225 @@
+                blockEntityTicker.tick(this.getLevel(), this.getPos(), blockEntity.getBlockState(), blockEntity);
+            }
+    }
+
+    private boolean canAccelerate(BlockEntity blockEntity){
+        if(blockEntity instanceof PipeBlockEntity || blockEntity instanceof IMachineBlockEntity) return false;
+
+        generateWorldAcceleratorBlacklist();
+
+        final Class<? extends BlockEntity> blockEntityClass = blockEntity.getClass();
+        if (blacklistCache.containsKey(blockEntityClass)) {
+            return blacklistCache.getBoolean(blockEntityClass);
+        }
+
+        for (Class<?> clazz : blacklistedClasses.values()) {
+            if (clazz.isAssignableFrom(blockEntityClass)) {
+                // Is a subclass, so it cannot be accelerated
+                blacklistCache.put(blockEntityClass, false);
+                return false;
+            }
+        }
+
+        blacklistCache.put(blockEntityClass, true);
+        return true;
+    }
+
+    @Override
+    public void onLoad() {
+        super.onLoad();
+        subscribeServerTick(this::update);
+    }
+
+    @Override
+    protected @NotNull InteractionResult onScrewdriverClick(Player playerIn, InteractionHand hand, Direction gridSide, BlockHitResult hitResult) {
+        if (!isRemote()) {
+            isRandomTickMode = !isRandomTickMode;
+            playerIn.sendSystemMessage(Component.translatable(isRandomTickMode?"gtceu.machine.world_accelerator.mode_entity":"gtceu.machine.world_accelerator.mode_tile"));
+            scheduleRenderUpdate();
+        }
+        return InteractionResult.CONSUME;
+    }
+
+    @Override
+    public boolean isWorkingEnabled() {
+        return this.isWorkingEnabled;
+    }
+
+    @Override
+    public void setWorkingEnabled(boolean isWorkingAllowed) {
+        if(isWorkingAllowed==this.isWorkingEnabled) return;
+        this.isWorkingEnabled = isWorkingAllowed;
+        scheduleRenderUpdate();
+    }
+
+    @Override
+    public boolean isActive() {
+        return active;
+    }
+
+    public void setActive(boolean active) {
+        if(active==this.active) return;
+        this.active = active;
+        scheduleRenderUpdate();
+    }
+
+    private static void generateWorldAcceleratorBlacklist(){
+        if (!gatheredClasses) {
+            for (String name : ConfigHolder.INSTANCE.machines.worldAcceleratorBlacklist) {
+                if (!blacklistedClasses.containsKey(name)) {
+                    try {
+                        blacklistedClasses.put(name, Class.forName(name));
+                    } catch (ClassNotFoundException ignored) {
+                        GTCEu.LOGGER.warn("Could not find class {} for World Accelerator Blacklist!", name);
+                    }
+                }
+            }
+
+            try {
+                // Block CoFH tile entities by default, non-overridable
+                String cofhTileClass = "cofh.thermalexpansion.block.device.TileDeviceBase";
```

Comment:
did you check that this class name is still correct?

Reply:
Actually no lemme check